check_run/check_run/doctype/check_run/check_run.py

```diff
@@ -46,6 +46,15 @@ def validate(self):
 				self.set_default_dates()
 		else:
 			self.validate_last_check_number()
+		# Check all selected invoices have correct docstatus (saved/submitted)
+		selected = [txn for txn in json.loads(self.get('transactions')) if txn['pay']]
+		wrong_status = []
+		for t in selected:
+			doc = frappe.get_doc(t['doctype'], t['name'])
```

Comment:
Let's use `frappe.get_value` to speed this validation up, we don't need to load the whole document to check the status